logstash-core/lib/logstash/instrument/metric.rb

```diff
@@ -45,20 +57,14 @@ def time(key, &block)
       end
     end
 
-    # TODO
-    def self.create(name, collector = LogStash::Instrument::Collector.instance)
+    def self.create(namespace, collector = LogStash::Instrument::Collector.instance)
       reporter = LogStash::Instrument::Reporter::Stdout.new(collector)
-      Metric.new(collector, name)
+      Metric.new(collector, namespace)
     end
 
     private
-    def merge_keys(key)
-      valid_key!(key)
-      [@base_key, key.to_sym]
-    end
-    
-    def valid_key!(key)
-      raise MetricNoKeyProvided if key.nil? || key == ""
+    def validate_key!(key)
+      raise MetricNoKeyProvided if key.nil? || key.empty?
```

Comment:
You introduced here a validation method to check, for now only the presence of the key. I think if we keep this method, we should add some other checks, or change the name of the method to exist? or something like.

more validation could be done in the user level, but it all depends on how flexible/strict we aim to be, keeping in  mind possible errors.